### PR TITLE
Endpoint returning wallet-count

### PR DESCRIPTION
**Pull request by marcinwadon**

### CODE REVIEW

File: modules/core/src/main/scala/org/tessellation/domain/dag/DagService.scala [hunk 1]
```diff
@@ -10,4 +10,7 @@ trait DAGService[F[_]] {
 
   def getTotalSupply: F[Option[(BigInt, SnapshotOrdinal)]]
   def getTotalSupply(ordinal: SnapshotOrdinal): F[Option[(BigInt, SnapshotOrdinal)]]
+
+  def getTotalWallets: F[Option[(Int, SnapshotOrdinal)]]
```
Comment (TheMMaciek) [inline modules/core/src/main/scala/org/tessellation/domain/dag/DagService.scala:14]: ...`Count`. I would call it `getTotalWalletsCount`.

File: modules/core/src/main/scala/org/tessellation/http/routes/DagRoutes.scala [hunk 2]
```diff
@@ -38,6 +38,13 @@ final case class DagRoutes[F[_]: Async](dagService: DAGService[F], mkDagCell: L0
         case _ => NotFound()
       }
 
+    case GET -> Root / "total-wallets" =>
```
Comment (TheMMaciek) [inline modules/core/src/main/scala/org/tessellation/http/routes/DagRoutes.scala:41]: Maybe "total-wallets-count" or just "wallets-count" for me "total-wallets" sounds confusing.

File: modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala [hunk 3]
```diff
@@ -55,5 +55,20 @@ object DAGService {
 
       }
 
+      def getTotalWallets: F[Option[(Int, SnapshotOrdinal)]] =
+        globalSnapshotStorage.head.flatMap {
+          _.flatTraverse(snapshot => getTotalWallets(snapshot.value.ordinal))
```
Comment (TheMMaciek) [inline modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala:60]: You are fetching the same snapshot twice here. Maybe you should just extract the part retrieving needed fields into a separate function and use it in both `getTotalWallets` methods.

File: modules/core/src/main/scala/org/tessellation/http/routes/DagRoutes.scala (hunk 2 above)
Comment (tbekas) [reply modules/core/src/main/scala/org/tessellation/http/routes/DagRoutes.scala:41]: I think the more english version is
`user-count`
(singular noun)

File: modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala [hunk 4]
```diff
@@ -55,5 +55,20 @@ object DAGService {
...
+        }
+
+      def getTotalWallets(ordinal: SnapshotOrdinal): F[Option[(Int, SnapshotOrdinal)]] =
```
Comment (tbekas) [inline modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala:63]: This function receives ordinal and then returns it, which is odd. I think you don't need it at all. Once you have `head` you can do just `head.info.balances.size`.

**Review by tbekas — CHANGES_REQUESTED** [review]

File: modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala (hunk 4 above)
Comment (tbekas) [reply modules/core/src/main/scala/org/tessellation/infrastructure/dag/DAGService.scala]: Ok, I get it. Let's keep it then.

**Review by tbekas — APPROVED** [review]
LGTM

**Pull request merged**